src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,467 @@
+                        );
+
+        // Determine the amount of fees to bring back
+        uint256 fees = 0;
+        if (assetBalance > daiDebt) {
+            fees = assetBalance - daiDebt;
+
+            if (_add(amount, fees) > availableLiquidity) {
+                // Don't need safe-math because this is constrained above
+                fees = availableLiquidity - amount;
+            }
+        }
+
+        if (amount == 0 && fees == 0) {
+            emit Unwind(ilk, 0);
+            return;
+        }
+
+        require(amount <= 2 ** 255, "DssDirectDepositHub/overflow");
+
+        // To save gas you can bring the fees back with the unwind
+        uint256 total = _add(amount, fees);
+        pool.withdraw(total);
+        daiJoin.join(address(pool), total);
+
+        // normalized debt == erc20 DAI to pool (Vat rate for this ilk fixed to 1 RAY)
+
+        if (mode == Mode.NORMAL) {
+            vat.frob(ilk, address(pool), address(pool), address(pool), -int256(amount), -int256(amount));
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(pool), vow, _mul(fees, RAY));
+        } else if (mode == Mode.MODULE_CULLED) {
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(pool), vow, _mul(total, RAY));
+        } else {
+            // This can be done with the assumption that the price of 1 aDai equals 1 DAI.
+            // That way we know that the prev End.skim call kept its gap[ilk] emptied as the CDP was always collateralized.
+            // Otherwise we couldn't just simply take away the collateral from the End module as the next line will be doing.
+            vat.slip(ilk, address(end), -int256(amount));
+            vat.move(address(pool), vow, _mul(total, RAY));
+        }
+
+        emit Unwind(ilk, amount);
+    }
+
+    function exec(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        uint256 availableAssets = pool.maxWithdraw();
+
+        if (vat.live() == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // MCD caged
+            require(end.debt() == 0, "DssDirectDepositHub/end-debt-already-set");
+            require(culled == 0, "DssDirectDepositHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                Mode.MCD_CAGED
+            );
+        } else if (live == 0) {
+            uint256 culled = ilks[ilk_].culled;
+            // This module caged
+            _unwind(
+                ilk_,
+                pool,
+                type(uint256).max,
+                availableAssets,
+                culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL
+            );
+        } else {
+            // Normal path
+            (uint256 totalAssets, uint256 targetAssets) = pool.calcSupplies(availableAssets);
```

Comment:
Very subjective so feel free to disgard. Maybe `totalAssets` => `currentAssets` is clearer.

Reply:
i think total assets makes more sense because it is meant to represent the total available assets and assets loaned out (in Aave it includes the stable debt and variable debt)